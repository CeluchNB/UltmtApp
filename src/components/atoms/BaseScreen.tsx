import React from 'react'
import { useTheme } from '../../hooks'
import { SafeAreaView, StyleSheet, View } from 'react-native'

interface BaseScreenProps {
    containerWidth: string
    children?: React.ReactNode
}

const BaseScreen: React.FC<BaseScreenProps> = props => {
    const { containerWidth, children } = props
    const {
        theme: { colors },
    } = useTheme()

    const styles = StyleSheet.create({
        screen: {
            height: '100%',
            backgroundColor: colors.primary,
        },
        container: {
            width: containerWidth,
            alignSelf: 'center',
        },
    })
    return (
        <SafeAreaView style={styles.screen}>
            <View style={styles.container}>{children}</View>
        </SafeAreaView>
    )
}

export default BaseScreen
