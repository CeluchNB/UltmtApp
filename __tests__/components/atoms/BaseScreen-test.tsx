import BaseScreen from '../../../src/components/atoms/BaseScreen'
import React from 'react'
import { View } from 'react-native'
import { render } from '@testing-library/react-native'

it('should match snapshot', () => {
    const snaphot = render(
        <BaseScreen containerWidth="80%">
            <View />
        </BaseScreen>,
    )

    expect(snaphot.toJSON()).toMatchSnapshot()
})
