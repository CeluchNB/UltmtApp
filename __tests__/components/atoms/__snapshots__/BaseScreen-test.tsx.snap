// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "backgroundColor": "#303030",
      "height": "100%",
    }
  }
>
  <View
    style={
      Object {
        "alignSelf": "center",
        "width": "80%",
      }
    }
  >
    <View />
  </View>
</RCTSafeAreaView>
`;
